public/js/src/appstore.tag

```diff
@@ -257,43 +259,119 @@ function AppStore(host, services) {
         });
     });
 
+    self.on('bridge_update_changes', function (newChanges) {
+        changes = newChanges;
+        RiotControl.trigger('bridge_have_changes');
+    });
+
+    self.on('bridge_have_changes', function () {
+        d('BridgeStore::bridge_have_changes', changes.length > 0);
+        RiotControl.trigger('bridge_have_changes_result', changes.length > 0);
+    });
+
+    self.on('bridge_updated_changes', function () {
+        d('BridgeStore::bridge_updated_changes');
+        RiotControl.trigger('bridge_have_changes');
+    });
+
+    self.on('show_changes_panel', function () {
+        d('BridgeStore::show_changes_panel')
+
+        RiotControl.trigger('open_changes_modal', changes);
+    });
+
+    self.on('shipit_save_changes', function () {
+        d('ShipitStore::shipit_save_changes', changes);
+        if (changes.length > 0) {
+            var x = changes.length;
+            RiotControl.trigger('flash_message', 'message', 'Saving x changes'.replace('x', x));
+
+            /*
+             * Using jQuery.when; which is like Promise.all, but works with jQuery.ajax out of the box
+             * One issue though, is that Promise.all takes an array of Promises, whereas jQuery.when
+             * takes each Deferred/Promise as a argument: ([x, y, z]) vs (x, y, z)
+             * So, I'm using Function.apply to map the Deferred objects from an array to arguments
+             * The changes.map call creates each jQuery.ajax call
+             */
+            $.when.apply(
+                this,
+                changes.map(function (change) {
+                    return $.ajax({
+                        method: change.method,
+                        url: change.url,
+                        dataType: 'json',
+                        contentType: 'application/json; charset=utf-8',
+                        accepts: 'application/json',
+                        data: JSON.stringify(change.data),
+                        headers: change.headers
+                    });
+                })
+            )
+            .then(
+                function () {
+                    d('ShipitStore::shipit_save_changes::when::ajax::then', arguments);
+                    RiotControl.trigger('flash_message', 'success', 'Changes saved')
+                    changes = [];
+                    RiotControl.trigger('bridge_updated_changes');
+                },
+                function () {
+                    d('ShipitStore::shipit_save_changes::when::ajax::catch', arguments);
+                    RiotControl.trigger('flash_message', 'error', 'Changes failed!');
+                    // TODO: Need to determine what to do with the changes array
+                    // ??? RiotControl.trigger('bridge_updated_changes');
```

Comment:
I'll look into removing the successful calls. That's a great idea. 

The error handler returns all the results for each call. So, I might be able to hook into that. I'll work some more on that.